firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -173,15 +191,16 @@ static AppStartTrace getInstance(TransportManager transportManager, Clock clock)
     this.clock = clock;
     this.configResolver = configResolver;
     this.executorService = executorService;
-
-    StartupTime startupTime = FirebaseApp.getInstance().get(StartupTime.class);
-    if (startupTime == null) {
-      firebaseStartupTime = new Timer();
-    } else {
-      firebaseStartupTime =
-          Timer.ofElapsedRealtime(startupTime.getElapsedRealtime(), startupTime.getUptimeMillis());
-    }
     this.experimentTtid = TraceMetric.newBuilder().setName("_experiment_app_start_ttid");
+    // Set the timestamp for process-start (beginning of BIND_APPLICATION), if available
+    this.processStartTime =
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
+            ? Timer.ofElapsedRealtime(Process.getStartElapsedRealtime())
+            : null;
+    // Set the timestamp for Firebase's first class's class-loading (approx.), if available
+    StartupTime firebaseStart = FirebaseApp.getInstance().get(StartupTime.class);
+    this.firebaseClassLoadTime =
+        firebaseStart != null ? Timer.ofElapsedRealtime(firebaseStart.getElapsedRealtime()) : null;
```

Comment:
NIT: Would firebaseStart ever be NULL? Since it is classLoad, it should not be NULL I presume? Is this check for completeness?

Reply:
According to Acore, it could be null when developers disable the ContentProvider (not documented but it's possible in Android). This should be extremely rare but we still need to cover the null case.